### ISSUE TITLE
Ubuntu - Decompressed folder is left without any permissions 0000

### ISSUE THREAD

**Issue by roeeyud** [kevva/decompress#24]
When I run my server in Ubuntu (Ubuntu 14.04 LTS) the decompressed destination folder is left without any permissions. 

var decompress = new Decompress({mode: '755'});
decompress.src(file).dest(destFolder).use(Decompress.zip).run(function (err) {
             if (err)
                    throw err;
});

I get a Error: EACCES, open <certine files decompress is trying to move to destFolder>

When I check destFolder I find it was created without any permissions (octal: 0000).

This forces me to run as sudo which really isn't ideal.
Please help.

**Comment by shinnn** [comment]
For now you can use [the `mode` option](https://github.com/kevva/decompress#optionsmode) to fix file permissions.

**Comment by roeeyud** [comment]
I use '755' when calling Decompress.
Still didn't work.

**Comment by headcr4sh** [comment]
I have encountered the same issue here. Permissions are always set to '000' and the "mode" option doesn't seem to work.

**Comment by kevva** [comment]
I am redoing this module so expect a fix soon-ish.

**Comment by piranna** [comment]
I have the same problem when downloading a zip file (if this helps):

```Javascript
const GENFATFS_URL="https://github.com/NodeOS/genfatfs/archive/master.zip"
const SRC_DIR='deps/genfatfs'

  Download({ extract: true, strip: 1 })
  .get(GENFATFS_URL, SRC_DIR)
//  .use(progress())
  .run(function(error)
  {
    ...
  ])
```

**Comment by joeljeske** [comment]
I am also getting this error. It is fixed by running as `sudo` but this isn't good. 

+1

**Comment by AndreasPizsa** [comment]
+1

**Comment by einfallstoll** [comment]
Since `Decompress` seems to be the only nice unzipper for node.js, I built a small workaround using `chmodr` from isaacs:

```
new Decompress()
.src(sourcePath)
.dest(destinationPath)
.use(Decompress.zip({
    strip: 1
}))
.run(function(error) {
    if (error) {
        console.error(error);
    } else {
        chmodr(destinationPath, 0644, callback);
    }
});
```

**Comment by FokkeZB** [comment]
@einfallstoll that doesn't work (for me) since run itself will fail with an EACCESS if the zip contains files in subdirectories.

**Comment by FokkeZB** [comment]
Patching `decompress-unzip` (3.3.0) with https://github.com/kevva/decompress-unzip/pull/10 fixes it. 

Please @kevva get it merged and do a patch release of decompress with the updated unzip.

**Comment by einfallstoll** [comment]
@FokkeZB Oh, I see. I only decompressed flat zips so far.

So far it should work to `npm install TrySound/decompress-unzip --save` as you mentioned.

**Comment by jlines** [comment]
+1

**Comment by stefanmaric** [comment]
Having the same issue. Trying to unzip a file on Ubuntu gives a `EACCES` error.

**Comment by guy-mograbi-at-gigaspaces** [comment]
+1 me too.. very urgent, last sprint.. :(

**Comment by w33ble** [comment]
Just ran in to this on OSX too. Even with using a flat `zip`, the files it writes are all permission 000.

**Comment by kevva** [comment]
Should be fixed in `4.0.0`. Let me know if it isn't.

**Comment by SamVerschueren** [comment]
'''‎''''''                                                                                                                                                                                                   Sent from my BlackBerry 10 smartphone.'                                                                                                                                                                                                                From: Kevin MårtenssonSent: Friday 20 May 2016 17:30To: kevva/decompres'''sReply To: kevva/deco'mpressSubject: Re: [':'kevva/decompress] Ubuntu - Decompresse'd folder is left without any permissio'n''')'','''''''''''''€s 0000 .;3/,€6'956££:$`'`'$$'``,£$`''''(#24')‎''',''''''''''''‎'''''
